### PR TITLE
added function for ordering roles of members

### LINKED ISSUE #70: Ordering of Roles under some projects
## Description

Referring https://github.com/creativecommons/creativecommons.github.io-source/issues/481
On the Community Team page, the ordering of the roles under CC Catalog API and Community Building Teams seems to be in reverse order than all others (This is only the case with these two categories).
It looks a little inconsistent with the other Projects ordered differently.

## Reproduction

Visit the Community Team page[https://opensource.creativecommons.org/community/community-team/members/]

## Expectation

The order should be reversed to have decreasing priority of roles.

## Screenshots

image

## Resolution

- [x] I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by ooyeraju**
## Fixes

Fixes #70  by @ooyeraju

## Description
Ordered roles of members under projects 

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: push_data_to_ccos/get_community_team_data.py [hunk 1]
```diff
@@ -94,6 +95,33 @@ def generate_databag():
...
+    community_builders_priority = {
+        "Community Collaborator": 1, "Community Contributor": 2}
+
+    for first_order_key in databag:
+        if first_order_key == 'projects':
+            for i in range(len(databag['projects'])):
+                member = databag['projects'][i]['members']
+                name_sorted_member = sorted(member, key=lambda x: x['name'])
+                databag['projects'][i]['members'] = sorted(
+                    name_sorted_member, key=lambda x: project_priority[x['role']])
```
Comment (dhruvkb) [inline push_data_to_ccos/get_community_team_data.py:113]: Can use `sort()` to sort the lists in place?

File: push_data_to_ccos/get_community_team_data.py [hunk 2]
```diff
@@ -94,6 +95,33 @@ def generate_databag():
...
+    This function orderes the member according to their roles
+    """
+    project_priority = {"Project Maintainer": 1, "Project Core Committer": 2,
+                        "Project Collaborator": 3, "Project Contributor": 4, }
+    community_builders_priority = {
+        "Community Collaborator": 1, "Community Contributor": 2}
+
+    for first_order_key in databag:
+        if first_order_key == 'projects':
+            for i in range(len(databag['projects'])):
```
Comment (dhruvkb) [inline push_data_to_ccos/get_community_team_data.py:109]: Using the for-in loop would be more Pythonic as
```suggestion
            for project in databag['projects']:
```

**Review by dhruvkb — CHANGES_REQUESTED** [review]
A couple of additional changes: 
- Lint with [Black](https://pypi.org/project/black/) as line length seems to exceed the 79 char limit
- Place `sort_databag()` under `prune_databag()` to respect order in which they are invoked

File: push_data_to_ccos/get_community_team_data.py (hunk 1 above)
Comment (ooyeraju) [reply push_data_to_ccos/get_community_team_data.py]: yes, I have done the changes

File: push_data_to_ccos/get_community_team_data.py (hunk 2 above)
Comment (ooyeraju) [reply push_data_to_ccos/get_community_team_data.py]: thanks, done as you said.

**Review by dhruvkb — APPROVED** [review]
Looks good to me!

**Comment by ooyeraju** [comment]
Thanks a lot @dhruvkb for the guidence.

**Comment by dhruvkb** [comment]
@ooyeraju you did good work!

**Pull request merged**